main.tf

```
@@ -1,13 +1,69 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled      = module.this.enabled
+  cluster_name = join("", aws_ecs_cluster.default.*.name)
+
+  capacity_providers_fargate = [for name, is_enabled in {
+    FARGATE : var.capacity_providers_fargate,
+    FARGATE_SPOT : var.capacity_providers_fargate_spot
+  } : name if is_enabled]
+
+  capacity_providers = distinct(concat(
+    [for key, value in aws_ecs_capacity_provider.ec2 : value.name],
```

This should work (please test) and is more readable

```suggestion
    aws_ecs_capacity_provider.ec2[*].name,
```